k6error/internal.go

```diff
@@ -0,0 +1,42 @@
+package k6error
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Internal represents an internal error which
+// should abort the whole test run. These sorts
+// of errors should be used when we want to
+// signal to the maintainer of the project that
+// something unexpected has happened.
+type Internal struct{ err error }
+
+// AsInternal wraps the given error into an
+// Internal error type.
+func AsInternal(err error) *Internal { return &Internal{err: fmt.Errorf("internal: %w", err)} }
+
+// AsInternal wraps the given msg into an
+// Internal error type.
+func AsInternalError(msg string) *Internal {
+	return &Internal{err: fmt.Errorf("internal: %w", errors.New(msg))}
+}
+
+// Error will return the msg of the wrapped error.
+func (i *Internal) Error() string { return i.err.Error() }
+
+// Internal will return true to signal that it
+// is an internal error.
+func (i *Internal) Internal() bool { return true }
+
+type internal interface {
+	Internal() bool
+}
+
+// IsInternal returns true if err is of type
+// Internal.
+func IsInternal(err error) bool {
```

Comment:
Same consideration as in https://github.com/grafana/xk6-browser/pull/744/commits/e4aba5550f1fcad42ba0b0773a893313b150339e#r1093114677.